var/spack/repos/builtin/packages/scotch/package.py

```diff
@@ -37,13 +37,18 @@ class Scotch(Package):
     list_url = "http://gforge.inria.fr/frs/?group_id=248"
 
     version('6.0.3', '10b0cc0f184de2de99859eafaca83cfc')
-    version('6.0.0', 'c50d6187462ba801f9a82133ee666e8e')
-    version('5.1.10b', 'f587201d6cf5cf63527182fbfba70753')
+    version('6.0.0', 'ba117428c0a6cd97d0c93e8b872bb3fe')
+    version('5.1.10b', '9b8622b39c141ecaca4a46298486fd99')
+
+    # This must be set for version <= 6.0.0
+    md5_esmumps = { Version('5.1.10b'): 'f587201d6cf5cf63527182fbfba70753',
+                    Version('6.0.0'): 'c50d6187462ba801f9a82133ee666e8e' }
```

Comment:
Based on what I discovered while working on my upgrades to this package, I don't think that it's worth considering the `~esmumps` archive variants of older `scotch` versions.  The `+esmumps` archive variants are identical to their `~esmumps` counterparts just with an additional `esmumps` subdirectory and additional build flags, so they seem preferable to use in all cases.  Is there any particular reason why you'd like to have both supported?